### PR TITLE
Add warnings for future breaking changes in Az.Resources policy CRUD cmdlets

### PR DESCRIPTION
**Pull request by mentat9**
## Description
Add breaking change warnings to Azure Policy CRUD cmdlets in Az.Resources module.

## Checklist

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAssignment.cs [hunk 1]
```diff
@@ -27,6 +27,9 @@ namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
     /// <summary>
     /// Gets the policy assignment.
     /// </summary>
+    [CmdletOutputBreakingChangeWithVersion(
+        typeof(PsPolicyAssignment), deprecateByAzVersion: "11.0.0", deprecateByVersion: "6.9.2", DeprecatedOutputProperties = new[] { "Properties" },
```
Comment (YanaXu) [inline src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAssignment.cs:31]: the current module version is 6.9.1, the next major version should be 7.0.0.

Comment (mentat9) [reply src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAssignment.cs]: Fixed, thanks.

File: src/Resources/Resources/ChangeLog.md [hunk 2]
```diff
@@ -20,6 +20,20 @@
 
 ## Upcoming Release
 * Implemented logic that allows Deployment Stack objects to be piped into Save and Remove Deployment Stack cmdlets.
+### `New-AzPolicyAssignment`, `Get-AzPolicyAssignment`, `Set-AzPolicyAssignment`
```
Comment (YanaXu) [inline src/Resources/Resources/ChangeLog.md:23]: This changelog is not correct. Please update it.

**Comment by YanaXu** [comment]
Hi @mentat9 , I'll close this PR since #22691 has been merged.

**Pull request closed without merging**